src/main/java/com/sinch/xms/api/MoSms.java

```diff
@@ -28,10 +28,10 @@
 
 /**
  * Base class for mobile originated messages. A mobile originated message can have either a {@link
- * MoTextSms textual} or a {@link MoBinarySms binary} message payload.
+ * MoTextSms textual} or a {@link MoBinarySms binary} or a {@link MoMms} message payload.
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
-@JsonSubTypes({@Type(MoTextSms.class), @Type(MoBinarySms.class)})
+@JsonSubTypes({@Type(MoTextSms.class), @Type(MoBinarySms.class), @Type(MoMms.class)})
```

Comment:
A philosophical question is if MoMms is a MoSms (the inheritance)

Reply:
It is not, but renaming MoSms => MoMessage might be more problematic I guess 🤷

Reply:
Yes, is simply Mo.class. Indeed it would break BC. I think your current solution might not break BC

Reply:
I prefer not to break BC

Reply:
Came up with a funnier name though MoXms